docker-compose-external-pg-ssl.yml

```diff
@@ -9,27 +6,30 @@ services:
     image: phusion/baseimage:focal-1.2.0
     volumes:
       - ./ssl:/certs
-    entrypoint: sh -c "apt install openssl && openssl genrsa -out /certs/ca.key 2048 && openssl req -x509 -new -nodes -key /certs/ca.key -sha256 -days 3650 -subj '/CN=CA' -out /certs/ca.crt"
+      - ./utils:/utils
+    entrypoint: sh -c "mkdir -p ./certs && apt install openssl && openssl genrsa -out /certs/ca.key 2048 && openssl req -x509 -new -nodes -key /certs/ca.key -sha256 -days 3650 -subj '/CN=CA' -out /certs/ca.crt && chown 1000 /certs/ca.crt"
 
   cert-generator:
     image: phusion/baseimage:focal-1.2.0
     depends_on:
       - ca-generator
     volumes:
       - ./ssl:/certs
-    entrypoint: sh -c "apt install openssl && openssl genrsa -out /certs/server.key 2048 && openssl req -new -key /certs/server.key -subj '/CN=localhost' -out /certs/server.csr && openssl x509 -req -in /certs/server.csr -CA /certs/ca.crt -CAkey /certs/ca.key -CAcreateserial -out /certs/server.crt -days 365 -sha256"
+      - ./utils:/utils
+    entrypoint: sh -c "apt install openssl && openssl genrsa -out /certs/server.key 2048 && openssl req -new -key /certs/server.key -config /utils/external-pgsql-ssl-config.cnf -out /certs/server.csr && openssl x509 -req -in /certs/server.csr -CA /certs/ca.crt -CAkey /certs/ca.key -CAcreateserial -out /certs/server.crt -days 365 -extensions v3_req -extfile /utils/external-pgsql-ssl-config.cnf && chown 1001 /certs/server.crt && chown 1001 /certs/server.key"
 
   client-cert-generator:
     image: phusion/baseimage:focal-1.2.0
     depends_on:
       - ca-generator
     volumes:
       - ./ssl:/certs
-    entrypoint: sh -c "apt install openssl && openssl genrsa -out /certs/client.key 2048 && openssl req -new -key /certs/client.key -subj '/CN=client' -out /certs/client.csr && openssl x509 -req -in /certs/client.csr -CA /certs/ca.crt -CAkey /certs/ca.key -CAcreateserial -out /certs/client.crt -days 365 -sha256"
+      - ./utils:/utils
+    entrypoint: sh -c "apt install openssl && openssl genrsa -out /certs/client.key 2048 && openssl req -new -key /certs/client.key -subj '/CN=client' -out /certs/client.csr && openssl x509 -req -in /certs/client.csr -CA /certs/ca.crt -CAkey /certs/ca.key -CAcreateserial -out /certs/client.crt -days 365 -sha256 && chown 1000 /certs/client.key && chown 1000 /certs/client.crt && chown 1000 /certs/ca.crt"
```

Comment:
Doesn't `apt` require the `-y` to assume yes during installation?

Reply:
yes, changed